lib/Backend/GlobOpt.cpp

```diff
@@ -4682,13 +4687,15 @@ GlobOpt::ValueNumberDst(IR::Instr **pInstr, Value *src1Val, Value *src2Val)
     case Js::OpCode::LdFld:
     case Js::OpCode::LdFldForTypeOf:
     case Js::OpCode::LdFldForCallApplyTarget:
-    // Do not transfer value type on ldFldForTypeOf to prevent copy-prop to LdRootFld in case the field doesn't exist since LdRootFldForTypeOf does not throw
+    // Do not transfer value type on LdRootFldForTypeOf to prevent copy-prop to LdRootFld in case the field doesn't exist since LdRootFldForTypeOf does not throw
     //case Js::OpCode::LdRootFldForTypeOf:
     case Js::OpCode::LdRootFld:
     case Js::OpCode::LdMethodFld:
     case Js::OpCode::LdRootMethodFld:
     case Js::OpCode::ScopedLdMethodFld:
     case Js::OpCode::LdMethodFromFlags:
+    case Js::OpCode::ScopedLdFld:
+    case Js::OpCode::ScopedLdFldForTypeOf:
```

Comment:
Should ScopedLdFldForTypeOf behave the same way as LdRootFldForTypeOf, given that we may wind up doing essentially that if the property is not in the scope chain?

Reply:
Hmm, yes. I'll change that. Thanks

---
In reply to: [211764618](https://github.com/Microsoft/ChakraCore/pull/5618#discussion_r211764618) [](ancestors = 211764618)